### PR TITLE
Initial integration with r2k for linux

### PR DESCRIPTION
**Pull request by P4N74**
For converting `unsigned long` to `ut64`, I will need to test it on 32-bit system first. If it runs ok, I will update the PR.

### CODE REVIEW

**Comment by RaminFP** [comment]
[link](url){:target="_blank"}

File: libr/io/p/io_r2k.c [hunk 1]
```diff
@@ -238,10 +239,91 @@ static int Init (const char * driverPath) {
 }
 #endif
 
+#if __linux__
+#include <sys/ioctl.h>
+
+struct r2k_data {
+	int pid;
+	unsigned long addr;
```
Comment (radare) [inline libr/io/p/io_r2k.c]: use ut64 here, userland doesnt knows if its talking to a 32 or 64bit kernel, so that must be casted down in kernel land

**Comment by radare** [comment]
can you do that test today? it's release time. the pr looks good to me

**Comment by P4N74** [comment]
I will work on it right now :) But without kernel module being available on radare2-extras, this might not be useful. Still I will test this now.

**Comment by radare** [comment]
should be easier to handler that from kernel land that way

thanks! looking forward this fix :) you can update extras after the release :)

**Comment by P4N74** [comment]
Updated code and it works reading/writing kernel memory is functioning properly.
I understood what you meant regarding `unsigned long` type. I will try to fix it asap.

**Comment by radare** [comment]
i still see no ut64

**Pull request merged**